### ISSUE TITLE
Decompiler: value-swap idiom in JsonElement.DeepEquals renders with synthetic temp name

### ISSUE THREAD

**Issue by richlander** [richlander/dotnet-inspect#3166]
## Defect

A value-swap idiom renders with a synthetic temp name:

```csharp
S_257 = element2;
element2 = element1;
element1 = S_257;
```

(The `NameEquals` local function has the equivalent with `S_256_1`.)

## Expected

Render the swap with a readable temp name (or recognized swap form). Lower
priority than the control-flow items in #3159, but tracked for completeness.

## Repro

```bash
dotnet-inspect System.Text.Json.JsonElement.DeepEquals -S "Decom*"
```

`Library: System.Text.Json | Source: Platform | TFM: net11.0`

Part of #3159.

**Comment by richlander** [comment]
Fixed by #3234 (merged to `main` at e849fa4e). The value-swap idiom now raises to the recognized tuple deconstruction swap `(q, p) = (p, q)` instead of a synthetic temp (`S_257 = ...`), guarded to distinct same-type by-value places and witnessed by the compile-back-Exact `CfgSampleClass.SwapStructPair` fixture. (The PR body's `Fixes/advances` phrasing didn't trigger auto-close.)
